CAPV manager crashed during the restoration of the mgmt cluster

/kind bug

**What steps did you take and what happened:**
The CAPV manager crashed during the restoration of the mgmt cluster with Velero 1.10+:
```
I0412 07:43:56.086037       1 vspheremachine_controller.go:185] "capv-controller-manager/vspheremachine-controller/default/wc-1-control-plane-jm6vv-4hhf6: Starting Reconcile VSphereMachine"
I0412 07:43:56.128007       1 controller.go:117] "Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference" controller="machine" controllerGroup="cluster.x-k8s.io" controllerKind="Machine" machine="default/wc-1-k888p-rvfjh" namespace="default" name="wc-1-k888p-rvfjh" reconcileID=b8e0218a-aea8-4477-95c7-898f1e80d6aa
panic: runtime error: invalid memory address or nil pointer dereference [recovered]panic: runtime error: invalid memory address or nil pointer dereference[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1a116cd]goroutine 818 [running]:sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:118 +0x1f4panic({0x1d27c60, 0x3600920})/usr/local/go/src/runtime/panic.go:838 +0x207sigs.k8s.io/cluster-api-provider-vsphere/controllers.nodeLabelReconciler.reconcileNormal({0xc0005a6540?, 0x22352e0?}, 0xc001485b20)/workspace/controllers/node_controller.go:134 +0x14dsigs.k8s.io/cluster-api-provider-vsphere/controllers.nodeLabelReconciler.Reconcile({0xc0005a6540?, 0x22352e0?}, {0x243bfe0, 0xc000ace360}, {{{0xc000d70526?, 0xc000df5cd8?}, {0xc000d70530?, 0x7fea56a93fb0?}}})/workspace/controllers/node_controller.go:129 +0x453sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x243bf38?, {0x243bfe0?, 0xc000ace360?}, {{{0xc000d70526?, 0x2012000?}, {0xc000d70530?, 0x4041f4?}}})/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:121 +0xc8sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000339220, {0x243bf38, 0xc00013ad80}, {0x1e47460?, 0xc001c0d280?})/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:320 +0x33csigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000339220, {0x243bf38, 0xc00013ad80})/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:273 +0x1d9sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:234 +0x85created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:230 +0x325 
```
This is caused by a bug that the code doesn't handle the error: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/controllers/node_controller.go#L113
 
During the restoration, Mchine objects are restored before the Clusters, so when the node controller is trying to get the Cluster, it will get the not found error.
The CAPV manager works well with Velero 1.9.x because we introduced a patch operation to restore the managedFields after restoring the object in Velero 1.10.x, this will trigger the reconciling logic of the node controller(only be triggered by the update event).


**What did you expect to happen:**
No such error shows up

**Anything else you would like to add:**
This fix is necessary for the `release-1.5` and `release-1.6` branches of CAPV.


**Environment:**

- Cluster-api-provider-vsphere version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

--- a/controllers/node_controller.go
+++ b/controllers/node_controller.go
@@ -111,11 +111,13 @@ func (r nodeLabelReconciler) Reconcile(ctx goctx.Context, req ctrl.Request) (_ c
 	}
 
 	cluster, err := clusterutilv1.GetClusterFromMetadata(r.ControllerContext, r.Client, machine.ObjectMeta)
-	if err == nil {
-		if annotations.IsPaused(cluster, machine) {
-			logger.V(4).Info("Machine linked to a cluster that is paused")
-			return reconcile.Result{}, nil
-		}
+	if err != nil {
+		logger.Error(err, "Could not fetch cluster data", "machine", key)
+		return reconcile.Result{}, nil
+	}
+	if annotations.IsPaused(cluster, machine) {
+		logger.V(4).Info("Machine linked to a cluster that is paused")
+		return reconcile.Result{}, nil
 	}
 
 	if !machine.DeletionTimestamp.IsZero() {

